components/autocomplete/completer-compat.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * This mod
+ */
+
+/**
+ * WordPress dependencies.
+ */
+import { deprecated } from '@wordpress/utils';
+
+const generateCompleterName = ( () => {
+	let count = 0;
+	return () => `backcompat-completer-${ count++ }`;
+} )();
+
+export function isDeprecatedCompleter( completer ) {
+	return 'onSelect' in completer;
+}
+
+export function toCompatibleCompleter( deprecatedCompleter ) {
+	deprecated( 'Original autocompleter interface', {
+		version: '2.8',
+		alternative: 'Latest autocompleter interface',
+		plugin: 'Gutenberg',
+		link: 'https://github.com/WordPress/gutenberg/blob/master/components/autocomplete/README.md',
+	} );
+
+	const optionalProperties = [ 'className', 'allowNode', 'allowContext' ]
```

Comment:
It needs to be converted to object, my fault :)

Reply:
I did the same thing. It's funny how our minds can just fill in the gaps sometimes. Thanks for fixing it.